### PR TITLE
[release-calendar] SearchBar Component

### PR DESCRIPTION
**Pull request by ajwhatson**
Handles that string from the input html form and brings it out to the App Component's State.

### CODE REVIEW

File: release-calendar/src/client/components/App.tsx [hunk 1]
```diff
@@ -16,7 +16,35 @@
...
+export class App extends React.Component<{}, AppState> {
+  constructor(props: unknown) {
+    super(props);
+    this.state = {
+      searchedValue: '',
+    };
+  }
+
+  handleSearch = (searchValue: string): void => {
+    console.log(this.state.searchedValue);
```
Comment (danielrozenberg) [inline release-calendar/src/client/components/App.tsx]: Debugging leftover? :D

File: release-calendar/src/client/components/App.tsx [hunk 2]
```diff
@@ -16,7 +16,35 @@
...
+  };
+
+  render(): JSX.Element {
+    return (
+      <div className='AMP-Release-Calender'>
+        <div className='AMP-Release-Calendar-Header'>
+          <Header title='AMP Release Calendar' />
+          <SearchBar handleSearch={this.handleSearch} />
+          <h4>{'this is what was searched: ' + this.state.searchedValue}</h4>
```
Comment (danielrozenberg) [inline release-calendar/src/client/components/App.tsx]: When a piece of text is hard coded, just write it outside of the `{ }` part:

```tsx
<h4>This is what was searched: {this.state.searchedValue}</h4>
```
(also I assume this is temporary code? add a TODO)

File: release-calendar/src/client/components/SearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import {ChangeEvent, FormEvent} from 'react';
+
+export interface SearchBarProps {
+  handleSearch: (searchValue: string) => void;
+}
+export class SearchBar extends React.Component<SearchBarProps, {}> {
+  constructor(props: Readonly<SearchBarProps>) {
+    super(props);
+    this.handleSearchValueChange = this.handleSearchValueChange.bind(this); //if I'm beinng honest, I'm not really
+    this.mySubmitHandler = this.mySubmitHandler.bind(this); //sure what the bindings actually accomplishes
```
Comment (danielrozenberg) [inline release-calendar/src/client/components/SearchBar.tsx]: So let's figure this out? :D

File: release-calendar/src/client/components/SearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  }
+  mySubmitHandler = (event: FormEvent<HTMLFormElement>): void => {
```
Comment (danielrozenberg) [inline release-calendar/src/client/components/SearchBar.tsx]: personal nit: you can just call these `submitHandler` and `searchValueChangeHandler`, `my` doesn't add anything to the name, and having them both use a consistent naming convention is nice

Comment (ajwhatson) [reply release-calendar/src/client/components/SearchBar.tsx]: haha, I didn't even realize I did that. Thanks for being so thorough. Will do!

**Pull request closed without merging**